File: python-pr/52/python/code-quality-readability-maintainability/user_manager_11.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
class UserManager:
    def __init__(self):
        self.users = []
    def add_user(self, name, email, age):
        if name and email and age and '@' in email and age > 0:
            self.users.append({'name': name, 'email': email, 'age': age})
            return True
        return False
    def get_user(self, email):
        for user in self.users:
            if user['email'] == email:
                return user
        return None
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/user_manager_11.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=1}
class UserManager {
    constructor() {
        this.users = [];
    }
    addUser(name, email, age) {
        if (name && email && age && email.includes('@') && age > 0) {
            this.users.push({name: name, email: email, age: age});
            return true;
        }
        return false;
    }
    getUser(email) {
        for (let user of this.users) {
            if (user.email === email) {
                return user;
            }
        }
        return null;
    }
}
// {/fact}